heareval/task_embeddings.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+"""
+Compute the embeddings for every task and store to disk.
+
+Since many tasks might be too large to store in GPU memory (or even
+CPU memory), and because Wavenet-like models will be expensive at
+inference time, we cache all embeddings to disk.
+
+One benefit of this approach is that since all embeddings are cached
+as numpy arrays, the final training code can be pytorch-only,
+regardless of whether the embedding model is tensorflow based.
+
+TODO:
+    * Ideally, we would run this within a docker container, for
+    security. https://github.com/neuralaudio/hear2021-eval-kit/issues/51
+    * Profiling should occur here (both embedding time AFTER loading
+    to GPU, and complete wall time include disk writes).
+    * This is currently pytorch only.
+    https://github.com/neuralaudio/hear2021-eval-kit/issues/52
+    Using the included get_audio_embedding_numpy, we could instead
+    have this work both for pytorch and tensorflow.
+    https://github.com/neuralaudio/hear2021-eval-kit/issues/49
+"""
+
+import csv
+import glob
+import os.path
+import pickle
+from importlib import import_module
+from typing import Any, Dict, Tuple
+
+import numpy as np
+import soundfile as sf
+import torch
+from torch.utils.data import DataLoader, Dataset
+from tqdm.auto import tqdm
+
+# This could instead be something from the participants
+EMBEDDING_PIP = "heareval.baseline"
```

Comment:
Command line arg?

Reply:
sure

Reply:
do you prefer 'click' or 'fire' ? 'click' seems slightly more active, I've only limited experience with both

Reply:
@jorshi I'm deferring doing this for now, since the module is global and I'm not sure whether we prefer to pass the module around as a parameter (seems cleaner?) or keep it global